lib/factory_bot_profiler/frame.rb

```diff
@@ -1,19 +1,35 @@
 module FactoryBotProfiler
   class Frame
-    attr_accessor :name, :start, :duration, :child_time
+    attr_accessor :name, :start, :duration, :child_time, :child_count
 
     def initialize(name)
       @name = name
       @start = Time.now
-      @child_time = 0
+      @child_time = Hash.new { 0 }
+      @child_count = Hash.new { 0 }
     end
 
     def observe_child(frame)
-      @child_time += frame.duration
+      @child_count[frame.name] += 1
+
+      @child_time[frame.name] += frame.self_time
+      frame.child_time.each do |factory_name, time|
+        @child_time[factory_name] += time
+      end
     end
 
     def finish!
       @duration = Time.now - @start
     end
+
+    def self_time
+      raise "self time called before frame was stopped" unless @duration
```

Comment:
The `duration` reader has this same problem. We could solve it in both places by tracking `@finished` and then moving this check to:

```rb
def duration
  raise "duration called before from was finished" unless @finished
  @duration
end
```